common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -341,8 +341,17 @@ public UTF8String substringSQL(int pos, int length) {
     // to the -ith element before the end of the sequence. If a start index i is 0, it
     // refers to the first element.
     int len = numChars();
+    // `len + pos` does not overflow as `len >= 0`.
```

Comment:
What happens if `len = 10` and `pos = Integer.MIN_VALUE`. I guess that `start` would have an incorrect value.

Reply:
The negative `pos` here `refers to the -ith element before the end of the sequence`, so if pos = Integer.MIN_VALUE, then the start should be `pos + len`. The final result of `EMPTY_UTF8` will be returned by `substring` when its param start and until are both negative. I also added a UT in 4dcfe81.

Reply:
I made misunderstaning. Thank you for clarification and adding a test.

Reply:
Thanks for reviewing!